app/src/main/java/org/schabi/newpipe/download/DownloadDialog.java

```diff
@@ -210,6 +231,17 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         initToolbar(view.findViewById(R.id.toolbar));
         setupDownloadOptions();
 
+        pathChange.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent i = new Intent(getContext(), FilePickerActivityHelper.class)
+                        .putExtra(FilePickerActivityHelper.EXTRA_ALLOW_MULTIPLE, false)
+                        .putExtra(FilePickerActivityHelper.EXTRA_ALLOW_CREATE_DIR, true)
+                        .putExtra(FilePickerActivityHelper.EXTRA_MODE, FilePickerActivityHelper.MODE_DIR);
+                startActivityForResult(i,change_Location_Code);
```

Comment:
You shouldn't just always use the no-nonsense file picker, but SAF or the no-nonsense file picker depending on what's set in the settings:

```suggestion
                Intent i;
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP
                        && NewPipeSettings.useStorageAccessFramework(ctx)) {
                    i = new Intent(Intent.ACTION_OPEN_DOCUMENT_TREE)
                            .putExtra("android.content.extra.SHOW_ADVANCED", true)
                            .addFlags(Intent.FLAG_GRANT_PERSISTABLE_URI_PERMISSION
                                    | StoredDirectoryHelper.PERMISSION_FLAGS);
                } else {
                    i = new Intent(getActivity(), FilePickerActivityHelper.class)
                            .putExtra(FilePickerActivityHelper.EXTRA_ALLOW_MULTIPLE, false)
                            .putExtra(FilePickerActivityHelper.EXTRA_ALLOW_CREATE_DIR, true)
                            .putExtra(FilePickerActivityHelper.EXTRA_MODE,
                                    FilePickerActivityHelper.MODE_DIR);
                }

                startActivityForResult(i, change_Location_Code);
```